File: solutions/clojure/077-longest-consecutive-sequence.clj
;; Problem 77: Longest Consecutive Sequence (LeetCode 128)
;; Difficulty: Med
;; Language: Clojure
;; 
(defn longest-consecutive [nums]
  (let [s (set nums)]
    (reduce (fn [mx n]
              (if (s (dec n)) mx
                (max mx (count (take-while s (iterate inc n))))))
            0 s)))

File: solutions/typescript/077-longest-consecutive-sequence.ts
/**
 * Problem 77: Longest Consecutive Sequence (LeetCode 128)
 * Difficulty: Med
 * Language: TypeScript
 */
function longestConsecutive(nums: number[]): number {
  const numSet = new Set(nums);
  return [...numSet].reduce((maxLen, n) => {
    if (numSet.has(n - 1)) return maxLen;
    const len = (function count(x: number): number {
      return numSet.has(x) ? 1 + count(x + 1) : 0;
    })(n);
    return Math.max(maxLen, len);
  }, 0);
}
